transform-plugins/src/test/java/co/cask/hydrator/plugin/ValueMapperTest.java

```diff
@@ -0,0 +1,715 @@
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.MapReduceManager;
+import co.cask.cdap.test.TestConfiguration;
+import co.cask.hydrator.common.MockPipelineConfigurer;
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.csv.CSVFormat;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test case for {@link ValueMapper}.
+ */
+
+public class ValueMapperTest extends HydratorTestBase {
+
+  private static final Schema SOURCE_SCHEMA =
+    Schema.recordOf("sourceRecord",
+                    Schema.Field.of(ValueMapperTest.ID, Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of(ValueMapperTest.NAME, Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of(ValueMapperTest.SALARY, Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of(ValueMapperTest.DESIGNATIONID,
+                                    Schema.unionOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.NULL))));
+
+  private static final String ID = "id";
+  private static final String NAME = "name";
+  private static final String SALARY = "salary";
+  private static final String DESIGNATIONID = "designationid";
+  private static final String DESIGNATIONNAME = "designationName";
+  private static final String SALARYDESC = "salaryDesc";
+
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration();
+
+  private static final ArtifactVersion CURRENT_VERSION = new ArtifactVersion("3.2.0");
+
+  private static final ArtifactId BATCH_APP_ARTIFACT_ID =
+          NamespaceId.DEFAULT.artifact("etlbatch", CURRENT_VERSION.getVersion());
+  private static final ArtifactSummary ETLBATCH_ARTIFACT =
+          new ArtifactSummary(BATCH_APP_ARTIFACT_ID.getArtifact(), BATCH_APP_ARTIFACT_ID.getVersion());
+
+  @BeforeClass
+  public static void setupTestClass() throws Exception {
+    // Add the ETL batch artifact and mock plugins.
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+
+    // Add our plugins artifact with the ETL batch artifact as its parent.
+    // This will make our plugins available to the ETL batch.
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("transform-plugins", "1.0.0"), BATCH_APP_ARTIFACT_ID,
+            ValueMapper.class, CSVFormat.class, Base64.class);
+  }
+
+  @Test
+  public void testSimpleFlow() throws Exception {
```

Comment:
testNull is a superset of this test.  If you already have testNull, this is not testing anything more and can be removed.